src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -251,6 +265,225 @@ private DocumentNode GenerateSqlGraphQLObjects(RuntimeEntities entities, Diction
             return new DocumentNode(nodes);
         }
 
+        /// <summary>
+        /// Helper method to traverse through all the relationships for all the entities exposed in the config.
+        /// For all the relationships defined in each entity's configuration, it adds an FK directive to all the
+        /// referencing fields of the referencing entity in the relationship. The values of such fields holding
+        /// foreign key references can come via insertions in the related entity. By adding ForiegnKeyDirective here,
+        /// we can later ensure that while creating input type for create mutations, these fields can be marked as
+        /// nullable/optional.
+        /// </summary>
+        /// <param name="objectTypes">Collection of object types.</param>
+        /// <param name="entities">Entities from runtime config.</param>
+        private void AddFKDirective(RuntimeEntities entities, Dictionary<string, ObjectTypeDefinitionNode> objectTypes)
+        {
+            foreach ((string sourceEntityName, ObjectTypeDefinitionNode sourceObjectTypeDefinitionNode) in objectTypes)
+            {
+                Entity entity = entities[sourceEntityName];
+                if (!entity.GraphQL.Enabled || entity.Relationships is null)
+                {
+                    continue;
+                }
+
+                string dataSourceName = _runtimeConfigProvider.GetConfig().GetDataSourceNameFromEntityName(sourceEntityName);
+                ISqlMetadataProvider sqlMetadataProvider = _metadataProviderFactory.GetMetadataProvider(dataSourceName);
+                SourceDefinition sourceDefinition = sqlMetadataProvider.GetSourceDefinition(sourceEntityName);
+                Dictionary<string, FieldDefinitionNode> sourceFieldDefinitions = sourceObjectTypeDefinitionNode.Fields.ToDictionary(field => field.Name.Value, field => field);
+
+                foreach((_, EntityRelationship relationship) in entity.Relationships)
+                {
+                    string targetEntityName = relationship.TargetEntity;
+                    if (!string.IsNullOrEmpty(relationship.LinkingObject))
+                    {
+                        // For M:N relationships, the fields in the entity are always referenced fields.
+                        continue;
+                    }
+
+                    if (// Retrieve all the relationship information for the source entity which is backed by this table definition
```

Comment:
super nit: Please move the comments out of the `if` condition block. It should be okay to have it immediately before or just inside the `if` condition